From: Gunnar Hjalmarsson <gunnarhj@ubuntu.com>
Date: Thu, 15 Feb 2018 21:21:58 +0100
Subject: Gettextize policy files

* Don't merge translations into the files
* Add gettext-domain="systemd" to description and message

Closes #8162, replaces #8118.

(cherry picked from commit 264d8dcc161e276d31dcde98a088d15cebbebbef)
---
 src/core/meson.build                             |  15 +--
 src/core/org.freedesktop.systemd1.policy.in.in   |  20 ++--
 src/hostname/meson.build                         |  10 +-
 src/hostname/org.freedesktop.hostname1.policy.in |  12 +--
 src/import/meson.build                           |  10 +-
 src/import/org.freedesktop.import1.policy.in     |  12 +--
 src/locale/meson.build                           |  10 +-
 src/locale/org.freedesktop.locale1.policy.in     |   8 +-
 src/login/meson.build                            |  10 +-
 src/login/org.freedesktop.login1.policy.in       | 128 +++++++++++------------
 src/machine/meson.build                          |  10 +-
 src/machine/org.freedesktop.machine1.policy.in   |  32 +++---
 src/resolve/meson.build                          |  10 +-
 src/resolve/org.freedesktop.resolve1.policy.in   |   8 +-
 src/timedate/meson.build                         |  10 +-
 src/timedate/org.freedesktop.timedate1.policy.in |  16 +--
 16 files changed, 150 insertions(+), 171 deletions(-)

diff --git a/src/core/meson.build b/src/core/meson.build
index bc03408..c58893b 100644
--- a/src/core/meson.build
+++ b/src/core/meson.build
@@ -211,19 +211,12 @@ install_data('org.freedesktop.systemd1.conf',
 install_data('org.freedesktop.systemd1.service',
              install_dir : dbussystemservicedir)
 
-policy_in = configure_file(
+policy = configure_file(
         input : 'org.freedesktop.systemd1.policy.in.in',
-        output : 'org.freedesktop.systemd1.policy.in',
-        configuration : substs)
-
-i18n.merge_file(
-        'org.freedesktop.systemd1.policy',
-        input : policy_in,
         output : 'org.freedesktop.systemd1.policy',
-        po_dir : po_dir,
-        data_dirs : po_dir,
-        install : install_polkit,
-        install_dir : polkitpolicydir)
+        configuration : substs)
+install_data(policy,
+             install_dir : polkitpolicydir)
 
 install_data('system.conf',
              'user.conf',
diff --git a/src/core/org.freedesktop.systemd1.policy.in.in b/src/core/org.freedesktop.systemd1.policy.in.in
index 2c6ed74..648221b 100644
--- a/src/core/org.freedesktop.systemd1.policy.in.in
+++ b/src/core/org.freedesktop.systemd1.policy.in.in
@@ -19,8 +19,8 @@
         <vendor_url>http://www.freedesktop.org/wiki/Software/systemd</vendor_url>
 
         <action id="org.freedesktop.systemd1.reply-password">
-                <description>Send passphrase back to system</description>
-                <message>Authentication is required to send the entered passphrase back to the system.</message>
+                <description gettext-domain="systemd">Send passphrase back to system</description>
+                <message gettext-domain="systemd">Authentication is required to send the entered passphrase back to the system.</message>
                 <defaults>
                         <allow_any>no</allow_any>
                         <allow_inactive>no</allow_inactive>
@@ -30,8 +30,8 @@
         </action>
 
         <action id="org.freedesktop.systemd1.manage-units">
-                <description>Manage system services or other units</description>
-                <message>Authentication is required to manage system services or other units.</message>
+                <description gettext-domain="systemd">Manage system services or other units</description>
+                <message gettext-domain="systemd">Authentication is required to manage system services or other units.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -40,8 +40,8 @@
         </action>
 
         <action id="org.freedesktop.systemd1.manage-unit-files">
-                <description>Manage system service or unit files</description>
-                <message>Authentication is required to manage system service or unit files.</message>
+                <description gettext-domain="systemd">Manage system service or unit files</description>
+                <message gettext-domain="systemd">Authentication is required to manage system service or unit files.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -50,8 +50,8 @@
         </action>
 
         <action id="org.freedesktop.systemd1.set-environment">
-                <description>Set or unset system and service manager environment variables</description>
-                <message>Authentication is required to set or unset system and service manager environment variables.</message>
+                <description gettext-domain="systemd">Set or unset system and service manager environment variables</description>
+                <message gettext-domain="systemd">Authentication is required to set or unset system and service manager environment variables.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -60,8 +60,8 @@
         </action>
 
         <action id="org.freedesktop.systemd1.reload-daemon">
-                <description>Reload the systemd state</description>
-                <message>Authentication is required to reload the systemd state.</message>
+                <description gettext-domain="systemd">Reload the systemd state</description>
+                <message gettext-domain="systemd">Authentication is required to reload the systemd state.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
diff --git a/src/hostname/meson.build b/src/hostname/meson.build
index 75cc948..c35c668 100644
--- a/src/hostname/meson.build
+++ b/src/hostname/meson.build
@@ -21,12 +21,10 @@ if conf.get('ENABLE_HOSTNAMED') == 1
         install_data('org.freedesktop.hostname1.service',
                      install_dir : dbussystemservicedir)
 
-        i18n.merge_file(
-                'org.freedesktop.hostname1.policy',
+        policy = configure_file(
                 input : 'org.freedesktop.hostname1.policy.in',
                 output : 'org.freedesktop.hostname1.policy',
-                po_dir : po_dir,
-                data_dirs : po_dir,
-                install : install_polkit,
-                install_dir : polkitpolicydir)
+                configuration : substs)
+        install_data(policy,
+                     install_dir : polkitpolicydir)
 endif
diff --git a/src/hostname/org.freedesktop.hostname1.policy.in b/src/hostname/org.freedesktop.hostname1.policy.in
index b10ca31..4ac82c6 100644
--- a/src/hostname/org.freedesktop.hostname1.policy.in
+++ b/src/hostname/org.freedesktop.hostname1.policy.in
@@ -19,8 +19,8 @@
         <vendor_url>http://www.freedesktop.org/wiki/Software/systemd</vendor_url>
 
         <action id="org.freedesktop.hostname1.set-hostname">
-                <description>Set host name</description>
-                <message>Authentication is required to set the local host name.</message>
+                <description gettext-domain="systemd">Set host name</description>
+                <message gettext-domain="systemd">Authentication is required to set the local host name.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -29,8 +29,8 @@
         </action>
 
         <action id="org.freedesktop.hostname1.set-static-hostname">
-                <description>Set static host name</description>
-                <message>Authentication is required to set the statically configured local host name, as well as the pretty host name.</message>
+                <description gettext-domain="systemd">Set static host name</description>
+                <message gettext-domain="systemd">Authentication is required to set the statically configured local host name, as well as the pretty host name.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -40,8 +40,8 @@
         </action>
 
         <action id="org.freedesktop.hostname1.set-machine-info">
-                <description>Set machine information</description>
-                <message>Authentication is required to set local machine information.</message>
+                <description gettext-domain="systemd">Set machine information</description>
+                <message gettext-domain="systemd">Authentication is required to set local machine information.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
diff --git a/src/import/meson.build b/src/import/meson.build
index 2dcc0bc..e5088b3 100644
--- a/src/import/meson.build
+++ b/src/import/meson.build
@@ -71,14 +71,12 @@ if conf.get('ENABLE_IMPORTD') == 1
         install_data('org.freedesktop.import1.service',
                      install_dir : dbussystemservicedir)
 
-        i18n.merge_file(
-                'org.freedesktop.import1.policy',
+        policy = configure_file(
                 input : 'org.freedesktop.import1.policy.in',
                 output : 'org.freedesktop.import1.policy',
-                po_dir : po_dir,
-                data_dirs : po_dir,
-                install : install_polkit,
-                install_dir : polkitpolicydir)
+                configuration : substs)
+        install_data(policy,
+                     install_dir : polkitpolicydir)
 
         install_data('import-pubring.gpg',
                      install_dir : rootlibexecdir)
diff --git a/src/import/org.freedesktop.import1.policy.in b/src/import/org.freedesktop.import1.policy.in
index d96ca2d..beea5fe 100644
--- a/src/import/org.freedesktop.import1.policy.in
+++ b/src/import/org.freedesktop.import1.policy.in
@@ -19,8 +19,8 @@
         <vendor_url>http://www.freedesktop.org/wiki/Software/systemd</vendor_url>
 
         <action id="org.freedesktop.import1.import">
-                <description>Import a VM or container image</description>
-                <message>Authentication is required to import a VM or container image</message>
+                <description gettext-domain="systemd">Import a VM or container image</description>
+                <message gettext-domain="systemd">Authentication is required to import a VM or container image</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -29,8 +29,8 @@
         </action>
 
         <action id="org.freedesktop.import1.export">
-                <description>Export a VM or container image</description>
-                <message>Authentication is required to export a VM or container image</message>
+                <description gettext-domain="systemd">Export a VM or container image</description>
+                <message gettext-domain="systemd">Authentication is required to export a VM or container image</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -39,8 +39,8 @@
         </action>
 
         <action id="org.freedesktop.import1.pull">
-                <description>Download a VM or container image</description>
-                <message>Authentication is required to download a VM or container image</message>
+                <description gettext-domain="systemd">Download a VM or container image</description>
+                <message gettext-domain="systemd">Authentication is required to download a VM or container image</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
diff --git a/src/locale/meson.build b/src/locale/meson.build
index dca2c51..30882cc 100644
--- a/src/locale/meson.build
+++ b/src/locale/meson.build
@@ -29,14 +29,12 @@ if conf.get('ENABLE_LOCALED') == 1
         install_data('org.freedesktop.locale1.service',
                      install_dir : dbussystemservicedir)
 
-        i18n.merge_file(
-                'org.freedesktop.locale1.policy',
+        policy = configure_file(
                 input : 'org.freedesktop.locale1.policy.in',
                 output : 'org.freedesktop.locale1.policy',
-                po_dir : po_dir,
-                data_dirs : po_dir,
-                install : install_polkit,
-                install_dir : polkitpolicydir)
+                configuration : substs)
+        install_data(policy,
+                     install_dir : polkitpolicydir)
 endif
 
 # If you know a way that allows the same variables to be used
diff --git a/src/locale/org.freedesktop.locale1.policy.in b/src/locale/org.freedesktop.locale1.policy.in
index 4c1c34d..f924174 100644
--- a/src/locale/org.freedesktop.locale1.policy.in
+++ b/src/locale/org.freedesktop.locale1.policy.in
@@ -19,8 +19,8 @@
         <vendor_url>http://www.freedesktop.org/wiki/Software/systemd</vendor_url>
 
         <action id="org.freedesktop.locale1.set-locale">
-                <description>Set system locale</description>
-                <message>Authentication is required to set the system locale.</message>
+                <description gettext-domain="systemd">Set system locale</description>
+                <message gettext-domain="systemd">Authentication is required to set the system locale.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -30,8 +30,8 @@
         </action>
 
         <action id="org.freedesktop.locale1.set-keyboard">
-                <description>Set system keyboard settings</description>
-                <message>Authentication is required to set the system keyboard settings.</message>
+                <description gettext-domain="systemd">Set system keyboard settings</description>
+                <message gettext-domain="systemd">Authentication is required to set the system keyboard settings.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
diff --git a/src/login/meson.build b/src/login/meson.build
index e8e4f7b..599c44e 100644
--- a/src/login/meson.build
+++ b/src/login/meson.build
@@ -88,14 +88,12 @@ if conf.get('ENABLE_LOGIND') == 1
         install_data('org.freedesktop.login1.service',
                      install_dir : dbussystemservicedir)
 
-        i18n.merge_file(
-                'org.freedesktop.login1.policy',
+        policy = configure_file(
                 input : 'org.freedesktop.login1.policy.in',
                 output : 'org.freedesktop.login1.policy',
-                po_dir : po_dir,
-                data_dirs : po_dir,
-                install : install_polkit,
-                install_dir : polkitpolicydir)
+                configuration : substs)
+        install_data(policy,
+                     install_dir : polkitpolicydir)
 
         install_data('70-power-switch.rules', install_dir : udevrulesdir)
 
diff --git a/src/login/org.freedesktop.login1.policy.in b/src/login/org.freedesktop.login1.policy.in
index 4716202..f1d1f95 100644
--- a/src/login/org.freedesktop.login1.policy.in
+++ b/src/login/org.freedesktop.login1.policy.in
@@ -19,8 +19,8 @@
         <vendor_url>http://www.freedesktop.org/wiki/Software/systemd</vendor_url>
 
         <action id="org.freedesktop.login1.inhibit-block-shutdown">
-                <description>Allow applications to inhibit system shutdown</description>
-                <message>Authentication is required for an application to inhibit system shutdown.</message>
+                <description gettext-domain="systemd">Allow applications to inhibit system shutdown</description>
+                <message gettext-domain="systemd">Authentication is required for an application to inhibit system shutdown.</message>
                 <defaults>
                         <allow_any>no</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -30,8 +30,8 @@
         </action>
 
         <action id="org.freedesktop.login1.inhibit-delay-shutdown">
-                <description>Allow applications to delay system shutdown</description>
-                <message>Authentication is required for an application to delay system shutdown.</message>
+                <description gettext-domain="systemd">Allow applications to delay system shutdown</description>
+                <message gettext-domain="systemd">Authentication is required for an application to delay system shutdown.</message>
                 <defaults>
                         <allow_any>yes</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -41,8 +41,8 @@
         </action>
 
         <action id="org.freedesktop.login1.inhibit-block-sleep">
-                <description>Allow applications to inhibit system sleep</description>
-                <message>Authentication is required for an application to inhibit system sleep.</message>
+                <description gettext-domain="systemd">Allow applications to inhibit system sleep</description>
+                <message gettext-domain="systemd">Authentication is required for an application to inhibit system sleep.</message>
                 <defaults>
                         <allow_any>no</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -52,8 +52,8 @@
         </action>
 
         <action id="org.freedesktop.login1.inhibit-delay-sleep">
-                <description>Allow applications to delay system sleep</description>
-                <message>Authentication is required for an application to delay system sleep.</message>
+                <description gettext-domain="systemd">Allow applications to delay system sleep</description>
+                <message gettext-domain="systemd">Authentication is required for an application to delay system sleep.</message>
                 <defaults>
                         <allow_any>yes</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -62,8 +62,8 @@
         </action>
 
         <action id="org.freedesktop.login1.inhibit-block-idle">
-                <description>Allow applications to inhibit automatic system suspend</description>
-                <message>Authentication is required for an application to inhibit automatic system suspend.</message>
+                <description gettext-domain="systemd">Allow applications to inhibit automatic system suspend</description>
+                <message gettext-domain="systemd">Authentication is required for an application to inhibit automatic system suspend.</message>
                 <defaults>
                         <allow_any>yes</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -72,8 +72,8 @@
         </action>
 
         <action id="org.freedesktop.login1.inhibit-handle-power-key">
-                <description>Allow applications to inhibit system handling of the power key</description>
-                <message>Authentication is required for an application to inhibit system handling of the power key.</message>
+                <description gettext-domain="systemd">Allow applications to inhibit system handling of the power key</description>
+                <message gettext-domain="systemd">Authentication is required for an application to inhibit system handling of the power key.</message>
                 <defaults>
                         <allow_any>no</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -83,8 +83,8 @@
         </action>
 
         <action id="org.freedesktop.login1.inhibit-handle-suspend-key">
-                <description>Allow applications to inhibit system handling of the suspend key</description>
-                <message>Authentication is required for an application to inhibit system handling of the suspend key.</message>
+                <description gettext-domain="systemd">Allow applications to inhibit system handling of the suspend key</description>
+                <message gettext-domain="systemd">Authentication is required for an application to inhibit system handling of the suspend key.</message>
                 <defaults>
                         <allow_any>no</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -94,8 +94,8 @@
         </action>
 
         <action id="org.freedesktop.login1.inhibit-handle-hibernate-key">
-                <description>Allow applications to inhibit system handling of the hibernate key</description>
-                <message>Authentication is required for an application to inhibit system handling of the hibernate key.</message>
+                <description gettext-domain="systemd">Allow applications to inhibit system handling of the hibernate key</description>
+                <message gettext-domain="systemd">Authentication is required for an application to inhibit system handling of the hibernate key.</message>
                 <defaults>
                         <allow_any>no</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -104,8 +104,8 @@
         </action>
 
         <action id="org.freedesktop.login1.inhibit-handle-lid-switch">
-                <description>Allow applications to inhibit system handling of the lid switch</description>
-                <message>Authentication is required for an application to inhibit system handling of the lid switch.</message>
+                <description gettext-domain="systemd">Allow applications to inhibit system handling of the lid switch</description>
+                <message gettext-domain="systemd">Authentication is required for an application to inhibit system handling of the lid switch.</message>
                 <defaults>
                         <allow_any>no</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -114,8 +114,8 @@
         </action>
 
         <action id="org.freedesktop.login1.set-self-linger">
-                <description>Allow non-logged-in user to run programs</description>
-                <message>Explicit request is required to run programs as a non-logged-in user.</message>
+                <description gettext-domain="systemd">Allow non-logged-in user to run programs</description>
+                <message gettext-domain="systemd">Explicit request is required to run programs as a non-logged-in user.</message>
                 <defaults>
                         <allow_any>yes</allow_any>
                         <allow_inactive>yes</allow_inactive>
@@ -124,8 +124,8 @@
         </action>
 
         <action id="org.freedesktop.login1.set-user-linger">
-                <description>Allow non-logged-in users to run programs</description>
-                <message>Authentication is required to run programs as a non-logged-in user.</message>
+                <description gettext-domain="systemd">Allow non-logged-in users to run programs</description>
+                <message gettext-domain="systemd">Authentication is required to run programs as a non-logged-in user.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -134,8 +134,8 @@
         </action>
 
         <action id="org.freedesktop.login1.attach-device">
-                <description>Allow attaching devices to seats</description>
-                <message>Authentication is required for attaching a device to a seat.</message>
+                <description gettext-domain="systemd">Allow attaching devices to seats</description>
+                <message gettext-domain="systemd">Authentication is required for attaching a device to a seat.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -145,8 +145,8 @@
         </action>
 
         <action id="org.freedesktop.login1.flush-devices">
-                <description>Flush device to seat attachments</description>
-                <message>Authentication is required for resetting how devices are attached to seats.</message>
+                <description gettext-domain="systemd">Flush device to seat attachments</description>
+                <message gettext-domain="systemd">Authentication is required for resetting how devices are attached to seats.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -155,8 +155,8 @@
         </action>
 
         <action id="org.freedesktop.login1.power-off">
-                <description>Power off the system</description>
-                <message>Authentication is required for powering off the system.</message>
+                <description gettext-domain="systemd">Power off the system</description>
+                <message gettext-domain="systemd">Authentication is required for powering off the system.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -166,8 +166,8 @@
         </action>
 
         <action id="org.freedesktop.login1.power-off-multiple-sessions">
-                <description>Power off the system while other users are logged in</description>
-                <message>Authentication is required for powering off the system while other users are logged in.</message>
+                <description gettext-domain="systemd">Power off the system while other users are logged in</description>
+                <message gettext-domain="systemd">Authentication is required for powering off the system while other users are logged in.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -177,8 +177,8 @@
         </action>
 
         <action id="org.freedesktop.login1.power-off-ignore-inhibit">
-                <description>Power off the system while an application asked to inhibit it</description>
-                <message>Authentication is required for powering off the system while an application asked to inhibit it.</message>
+                <description gettext-domain="systemd">Power off the system while an application asked to inhibit it</description>
+                <message gettext-domain="systemd">Authentication is required for powering off the system while an application asked to inhibit it.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -188,8 +188,8 @@
         </action>
 
         <action id="org.freedesktop.login1.reboot">
-                <description>Reboot the system</description>
-                <message>Authentication is required for rebooting the system.</message>
+                <description gettext-domain="systemd">Reboot the system</description>
+                <message gettext-domain="systemd">Authentication is required for rebooting the system.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -199,8 +199,8 @@
         </action>
 
         <action id="org.freedesktop.login1.reboot-multiple-sessions">
-                <description>Reboot the system while other users are logged in</description>
-                <message>Authentication is required for rebooting the system while other users are logged in.</message>
+                <description gettext-domain="systemd">Reboot the system while other users are logged in</description>
+                <message gettext-domain="systemd">Authentication is required for rebooting the system while other users are logged in.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -210,8 +210,8 @@
         </action>
 
         <action id="org.freedesktop.login1.reboot-ignore-inhibit">
-                <description>Reboot the system while an application asked to inhibit it</description>
-                <message>Authentication is required for rebooting the system while an application asked to inhibit it.</message>
+                <description gettext-domain="systemd">Reboot the system while an application asked to inhibit it</description>
+                <message gettext-domain="systemd">Authentication is required for rebooting the system while an application asked to inhibit it.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -221,8 +221,8 @@
         </action>
 
         <action id="org.freedesktop.login1.halt">
-                <description>Halt the system</description>
-                <message>Authentication is required for halting the system.</message>
+                <description gettext-domain="systemd">Halt the system</description>
+                <message gettext-domain="systemd">Authentication is required for halting the system.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -232,8 +232,8 @@
         </action>
 
         <action id="org.freedesktop.login1.halt-multiple-sessions">
-                <description>Halt the system while other users are logged in</description>
-                <message>Authentication is required for halting the system while other users are logged in.</message>
+                <description gettext-domain="systemd">Halt the system while other users are logged in</description>
+                <message gettext-domain="systemd">Authentication is required for halting the system while other users are logged in.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -243,8 +243,8 @@
         </action>
 
         <action id="org.freedesktop.login1.halt-ignore-inhibit">
-                <description>Halt the system while an application asked to inhibit it</description>
-                <message>Authentication is required for halting the system while an application asked to inhibit it.</message>
+                <description gettext-domain="systemd">Halt the system while an application asked to inhibit it</description>
+                <message gettext-domain="systemd">Authentication is required for halting the system while an application asked to inhibit it.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -254,8 +254,8 @@
         </action>
 
         <action id="org.freedesktop.login1.suspend">
-                <description>Suspend the system</description>
-                <message>Authentication is required for suspending the system.</message>
+                <description gettext-domain="systemd">Suspend the system</description>
+                <message gettext-domain="systemd">Authentication is required for suspending the system.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -264,8 +264,8 @@
         </action>
 
         <action id="org.freedesktop.login1.suspend-multiple-sessions">
-                <description>Suspend the system while other users are logged in</description>
-                <message>Authentication is required for suspending the system while other users are logged in.</message>
+                <description gettext-domain="systemd">Suspend the system while other users are logged in</description>
+                <message gettext-domain="systemd">Authentication is required for suspending the system while other users are logged in.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -275,8 +275,8 @@
         </action>
 
         <action id="org.freedesktop.login1.suspend-ignore-inhibit">
-                <description>Suspend the system while an application asked to inhibit it</description>
-                <message>Authentication is required for suspending the system while an application asked to inhibit it.</message>
+                <description gettext-domain="systemd">Suspend the system while an application asked to inhibit it</description>
+                <message gettext-domain="systemd">Authentication is required for suspending the system while an application asked to inhibit it.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -286,8 +286,8 @@
         </action>
 
         <action id="org.freedesktop.login1.hibernate">
-                <description>Hibernate the system</description>
-                <message>Authentication is required for hibernating the system.</message>
+                <description gettext-domain="systemd">Hibernate the system</description>
+                <message gettext-domain="systemd">Authentication is required for hibernating the system.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -296,8 +296,8 @@
         </action>
 
         <action id="org.freedesktop.login1.hibernate-multiple-sessions">
-                <description>Hibernate the system while other users are logged in</description>
-                <message>Authentication is required for hibernating the system while other users are logged in.</message>
+                <description gettext-domain="systemd">Hibernate the system while other users are logged in</description>
+                <message gettext-domain="systemd">Authentication is required for hibernating the system while other users are logged in.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -307,8 +307,8 @@
         </action>
 
         <action id="org.freedesktop.login1.hibernate-ignore-inhibit">
-                <description>Hibernate the system while an application asked to inhibit it</description>
-                <message>Authentication is required for hibernating the system while an application asked to inhibit it.</message>
+                <description gettext-domain="systemd">Hibernate the system while an application asked to inhibit it</description>
+                <message gettext-domain="systemd">Authentication is required for hibernating the system while an application asked to inhibit it.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -318,8 +318,8 @@
         </action>
 
         <action id="org.freedesktop.login1.manage">
-                <description>Manage active sessions, users and seats</description>
-                <message>Authentication is required for managing active sessions, users and seats.</message>
+                <description gettext-domain="systemd">Manage active sessions, users and seats</description>
+                <message gettext-domain="systemd">Authentication is required for managing active sessions, users and seats.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -328,8 +328,8 @@
         </action>
 
         <action id="org.freedesktop.login1.lock-sessions">
-                <description>Lock or unlock active sessions</description>
-                <message>Authentication is required to lock or unlock active sessions.</message>
+                <description gettext-domain="systemd">Lock or unlock active sessions</description>
+                <message gettext-domain="systemd">Authentication is required to lock or unlock active sessions.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -338,8 +338,8 @@
         </action>
 
         <action id="org.freedesktop.login1.set-reboot-to-firmware-setup">
-                <description>Allow indication to the firmware to boot to setup interface</description>
-                <message>Authentication is required to indicate to the firmware to boot to setup interface.</message>
+                <description gettext-domain="systemd">Allow indication to the firmware to boot to setup interface</description>
+                <message gettext-domain="systemd">Authentication is required to indicate to the firmware to boot to setup interface.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -348,8 +348,8 @@
         </action>
 
         <action id="org.freedesktop.login1.set-wall-message">
-                <description>Set a wall message</description>
-                <message>Authentication is required to set a wall message</message>
+                <description gettext-domain="systemd">Set a wall message</description>
+                <message gettext-domain="systemd">Authentication is required to set a wall message</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
diff --git a/src/machine/meson.build b/src/machine/meson.build
index 7ea5d9d..0f2944c 100644
--- a/src/machine/meson.build
+++ b/src/machine/meson.build
@@ -44,14 +44,12 @@ if conf.get('ENABLE_MACHINED') == 1
         install_data('org.freedesktop.machine1.service',
                      install_dir : dbussystemservicedir)
 
-        i18n.merge_file(
-                'org.freedesktop.machine1.policy',
+        policy = configure_file(
                 input : 'org.freedesktop.machine1.policy.in',
                 output : 'org.freedesktop.machine1.policy',
-                po_dir : po_dir,
-                data_dirs : po_dir,
-                install : install_polkit,
-                install_dir : polkitpolicydir)
+                configuration : substs)
+        install_data(policy,
+                     install_dir : polkitpolicydir)
 endif
 
 tests += [
diff --git a/src/machine/org.freedesktop.machine1.policy.in b/src/machine/org.freedesktop.machine1.policy.in
index eeeeb4c..039c3d4 100644
--- a/src/machine/org.freedesktop.machine1.policy.in
+++ b/src/machine/org.freedesktop.machine1.policy.in
@@ -19,8 +19,8 @@
         <vendor_url>http://www.freedesktop.org/wiki/Software/systemd</vendor_url>
 
         <action id="org.freedesktop.machine1.login">
-                <description>Log into a local container</description>
-                <message>Authentication is required to log into a local container.</message>
+                <description gettext-domain="systemd">Log into a local container</description>
+                <message gettext-domain="systemd">Authentication is required to log into a local container.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -29,8 +29,8 @@
         </action>
 
         <action id="org.freedesktop.machine1.host-login">
-                <description>Log into the local host</description>
-                <message>Authentication is required to log into the local host.</message>
+                <description gettext-domain="systemd">Log into the local host</description>
+                <message gettext-domain="systemd">Authentication is required to log into the local host.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -39,8 +39,8 @@
         </action>
 
         <action id="org.freedesktop.machine1.shell">
-                <description>Acquire a shell in a local container</description>
-                <message>Authentication is required to acquire a shell in a local container.</message>
+                <description gettext-domain="systemd">Acquire a shell in a local container</description>
+                <message gettext-domain="systemd">Authentication is required to acquire a shell in a local container.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -50,8 +50,8 @@
         </action>
 
         <action id="org.freedesktop.machine1.host-shell">
-                <description>Acquire a shell on the local host</description>
-                <message>Authentication is required to acquire a shell on the local host.</message>
+                <description gettext-domain="systemd">Acquire a shell on the local host</description>
+                <message gettext-domain="systemd">Authentication is required to acquire a shell on the local host.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -61,8 +61,8 @@
         </action>
 
         <action id="org.freedesktop.machine1.open-pty">
-                <description>Acquire a pseudo TTY in a local container</description>
-                <message>Authentication is required to acquire a pseudo TTY in a local container.</message>
+                <description gettext-domain="systemd">Acquire a pseudo TTY in a local container</description>
+                <message gettext-domain="systemd">Authentication is required to acquire a pseudo TTY in a local container.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -71,8 +71,8 @@
         </action>
 
         <action id="org.freedesktop.machine1.host-open-pty">
-                <description>Acquire a pseudo TTY on the local host</description>
-                <message>Authentication is required to acquire a pseudo TTY on the local host.</message>
+                <description gettext-domain="systemd">Acquire a pseudo TTY on the local host</description>
+                <message gettext-domain="systemd">Authentication is required to acquire a pseudo TTY on the local host.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -81,8 +81,8 @@
         </action>
 
         <action id="org.freedesktop.machine1.manage-machines">
-                <description>Manage local virtual machines and containers</description>
-                <message>Authentication is required to manage local virtual machines and containers.</message>
+                <description gettext-domain="systemd">Manage local virtual machines and containers</description>
+                <message gettext-domain="systemd">Authentication is required to manage local virtual machines and containers.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -92,8 +92,8 @@
         </action>
 
         <action id="org.freedesktop.machine1.manage-images">
-                <description>Manage local virtual machine and container images</description>
-                <message>Authentication is required to manage local virtual machine and container images.</message>
+                <description gettext-domain="systemd">Manage local virtual machine and container images</description>
+                <message gettext-domain="systemd">Authentication is required to manage local virtual machine and container images.</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
diff --git a/src/resolve/meson.build b/src/resolve/meson.build
index 15752d2..7e7876d 100644
--- a/src/resolve/meson.build
+++ b/src/resolve/meson.build
@@ -165,14 +165,12 @@ if conf.get('ENABLE_RESOLVE') == 1
         install_data('resolv.conf',
                      install_dir : rootlibexecdir)
 
-        i18n.merge_file(
-                'org.freedesktop.resolve1.policy',
+        policy = configure_file(
                 input : 'org.freedesktop.resolve1.policy.in',
                 output : 'org.freedesktop.resolve1.policy',
-                po_dir : po_dir,
-                data_dirs : po_dir,
-                install : install_polkit,
-                install_dir : polkitpolicydir)
+                configuration : substs)
+        install_data(policy,
+                     install_dir : polkitpolicydir)
 endif
 
 tests += [
diff --git a/src/resolve/org.freedesktop.resolve1.policy.in b/src/resolve/org.freedesktop.resolve1.policy.in
index da948eb..b65ba3e 100644
--- a/src/resolve/org.freedesktop.resolve1.policy.in
+++ b/src/resolve/org.freedesktop.resolve1.policy.in
@@ -19,8 +19,8 @@
         <vendor_url>http://www.freedesktop.org/wiki/Software/systemd</vendor_url>
 
         <action id="org.freedesktop.resolve1.register-service">
-                <description>Register a DNS-SD service</description>
-                <message>Authentication is required to register a DNS-SD service</message>
+                <description gettext-domain="systemd">Register a DNS-SD service</description>
+                <message gettext-domain="systemd">Authentication is required to register a DNS-SD service</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
@@ -30,8 +30,8 @@
         </action>
 
         <action id="org.freedesktop.resolve1.unregister-service">
-                <description>Unregister a DNS-SD service</description>
-                <message>Authentication is required to unregister a DNS-SD service</message>
+                <description gettext-domain="systemd">Unregister a DNS-SD service</description>
+                <message gettext-domain="systemd">Authentication is required to unregister a DNS-SD service</message>
                 <defaults>
                         <allow_any>auth_admin</allow_any>
                         <allow_inactive>auth_admin</allow_inactive>
diff --git a/src/timedate/meson.build b/src/timedate/meson.build
index 80e5cd2..6892596 100644
--- a/src/timedate/meson.build
+++ b/src/timedate/meson.build
@@ -21,12 +21,10 @@ if conf.get('ENABLE_TIMEDATED') == 1
         install_data('org.freedesktop.timedate1.service',
                      install_dir : dbussystemservicedir)
 
-        i18n.merge_file(
-                'org.freedesktop.timedate1.policy',
+        policy = configure_file(
                 input : 'org.freedesktop.timedate1.policy.in',
                 output : 'org.freedesktop.timedate1.policy',
-                po_dir : po_dir,
-                data_dirs : po_dir,
-                install : install_polkit,
-                install_dir : polkitpolicydir)
+                configuration : substs)
+        install_data(policy,
+                     install_dir : polkitpolicydir)
 endif
diff --git a/src/timedate/org.freedesktop.timedate1.policy.in b/src/timedate/org.freedesktop.timedate1.policy.in
index cc2e165..d488572 100644
--- a/src/timedate/org.freedesktop.timedate1.policy.in
+++ b/src/timedate/org.freedesktop.timedate1.policy.in
@@ -19,8 +19,8 @@
         <vendor_url>http://www.freedesktop.org/wiki/Software/systemd</vendor_url>
 
         <action id="org.freedesktop.timedate1.set-time">
-                <description>Set system time</description>
-                <message>Authentication is required to set the system time.</message>
+                <description gettext-domain="systemd">Set system time</description>
+                <message gettext-domain="systemd">Authentication is required to set the system time.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -30,8 +30,8 @@
         </action>
 
         <action id="org.freedesktop.timedate1.set-timezone">
-                <description>Set system timezone</description>
-                <message>Authentication is required to set the system timezone.</message>
+                <description gettext-domain="systemd">Set system timezone</description>
+                <message gettext-domain="systemd">Authentication is required to set the system timezone.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
                         <allow_inactive>auth_admin_keep</allow_inactive>
@@ -40,8 +40,8 @@
         </action>
 
         <action id="org.freedesktop.timedate1.set-local-rtc">
-                <description>Set RTC to local timezone or UTC</description>
-                <message>Authentication is required to control whether
+                <description gettext-domain="systemd">Set RTC to local timezone or UTC</description>
+                <message gettext-domain="systemd">Authentication is required to control whether
                 the RTC stores the local or UTC time.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
@@ -51,8 +51,8 @@
         </action>
 
         <action id="org.freedesktop.timedate1.set-ntp">
-                <description>Turn network time synchronization on or off</description>
-                <message>Authentication is required to control whether
+                <description gettext-domain="systemd">Turn network time synchronization on or off</description>
+                <message gettext-domain="systemd">Authentication is required to control whether
                 network time synchronization shall be enabled.</message>
                 <defaults>
                         <allow_any>auth_admin_keep</allow_any>
